vermouth/ffinput.py

```diff
@@ -51,196 +55,337 @@
+            ('link', ): self._new_link,
+            ('modification', ): self._new_modification,
+        }
+
+    def parse_header(self, line, lineno=0):
+        """
+        Parses a section header with line number `lineno`. Sets :attr:`section`
+        when applicable. Does not check whether `line` is a valid section
+        header.
+
+        Parameters
+        ----------
+        line: str
+        lineno: str
+
+        Returns
+        -------
+        object
+            The result of calling :meth:`finalize_section`, which is called
+            if a section ends.
+
+        Raises
+        ------
+        KeyError
+            If the section header is unknown.
+        """
+                        
+        prev_section = self.section
+
+        ended = []
+        section = self.section + [line.strip('[ ]').casefold()]
+        if tuple(section[-1:]) in self.METH_DICT:
+            self.section = section[-1:]
+        else:
+            while (tuple(section) not in self.METH_DICT
+                   and len(section) > 1):
+                ended.append(section.pop(-2))  # [a, b, c, d] -> [a, b, d]
+            self.section = section
+                          
+        result = None
+
+        if len(prev_section) != 0:
+            print(len(prev_section))
+            result = self.finalize_section(prev_section, ended)
+
+        action = self.header_actions.get(tuple(self.section))
+        if action:
+            action()
+
+        return result
+
+    def parse_section(self, line, lineno):
+        """
+        Parse `line` with line number `lineno` by looking up the section in
+        :attr:`METH_DICT` and calling that method.
+
+        On the contrary to the SectionLineParser, we can have wildcard sections.
+        Wildcard sections cannot have children sections.
+
+        Parameters
+        ----------
+        line: str
+        lineno: int
+
+        Returns
+        -------
+        object
+           The result returned by calling the registered method.
+        """
+        line = _substitute_macros(line, self.macros)
+        #wildcard_section = tuple(self.section[:-1] + ['*'])
+        end_section = []
+        if self.section:
+            end_section = self.section[-1]
+        if tuple(self.section) in self.METH_DICT:
+            method, kwargs = self.METH_DICT[tuple(self.section)]
+        #elif wildcard_section in self.METH_DICT:
+        #    method, kwargs = self.METH_DICT[wildcard_section]
+        else:
+            raise IOError("Can't parse line {} in section '{}' because the "
+                          "section is unknown".format(lineno, self.section))
+        try:
+            return method(self, line, lineno, **kwargs)
+        except Exception as error:
+            raise IOError("Problems parsing line {}. I think it should be a "
+                          "'{}' line, but I can't parse it as such."
+                          "".format(lineno, self.section)) from error
```

Comment:
In this case it could make sense to rewrite the parent method. I don't see directly why the parent is organized as it is. Also it is more a dummy method at the moment, which is only overwritten by the ffinput.

Reply:
@pckroon what do you think about reorganizing the parent method?

Reply:
AFAIK the parent method does the same as this one. The only difference i can spot is that this one does something with `end_section`, which it then discards.